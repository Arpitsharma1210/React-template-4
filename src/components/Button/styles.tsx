import { styled } from "styled-components";
import {
  fontWeight,
} from "../../theme/style.typography";


export const StyledButtonTextWrapper = styled.div<{ $bold?: string }>`
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 10px;
  font-weight: ${({ $bold }) =>
    $bold === "true" ? "bold" : fontWeight.regular};
`;
