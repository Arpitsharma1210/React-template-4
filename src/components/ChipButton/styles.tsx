import { styled, css } from "styled-components";
import {
  fontSize,
  baseFontFamily,
  fontWeight,
} from "../../theme/style.typography.js";
import { brand, colors } from "../../theme/style.palette.js";

type ButtonEnum = {
  [key: string]: {
    color: string;
    bgColor: string;
    borderColor: string;
  };
};

const Buttons: ButtonEnum = {
  yellow: {
    color: colors.yellow10,
    bgColor: brand.primaryMain,
    borderColor: brand.secondaryMain,
  },
  yellow2: {
    color: colors.yellow10,
    bgColor: colors.yellow50,
    borderColor: colors.yellow10,
  },
  green: {
    color: colors.green5,
    bgColor: colors.green25,
    borderColor: colors.green5,
  },
  blue: {
    color: colors.blue10,
    bgColor: colors.blue25,
    borderColor: colors.blue10,
  },
  red: {
    color: colors.red10,
    bgColor: colors.red50,
    borderColor: colors.red10,
  },
  orange: {
    color: colors.orange10,
    bgColor: colors.orange25,
    borderColor: colors.orange10,
  },
  white: {
    color: colors.grey100,
    bgColor: colors.white25,
    borderColor: colors.grey100,
  },
  purple: {
    color: colors.purple10,
    bgColor: colors.purple25,
    borderColor: colors.purple10,
  },
  grey:{
    color: colors.blackExtra,
    bgColor: colors.grey10,
    borderColor: colors.grey100,
  }
};

export const StyledButton = styled.button<{ color: string; size?: string }>`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  gap: 10px;

  cursor: pointer;
  border-radius: 4px;
  font-weight: ${fontWeight.regular};
  font-family: ${baseFontFamily};
  ${({ color }) => css`
    border: 1px solid ${Buttons[color].borderColor};
    color: ${Buttons[color].color};
    background-color: ${Buttons[color].bgColor};
  `}
  ${({ size }) =>
    (size === "sm" &&
      css`
        min-height: 23px;
        padding: 2px 6px;
        font-size: ${fontSize.b3};
        line-height: 16px;
      `) ||
    (size === "md" &&
      css`
        min-height: 28px;
        padding: 4px 12px;
        font-size: ${fontSize.b2};
        line-height: 19px;
      `)}
`;
