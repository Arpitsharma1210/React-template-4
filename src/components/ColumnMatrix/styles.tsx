import styled from "styled-components";
import { colors } from "../../theme/style.palette";
import { baseFontFamily, fontSize } from "../../theme/style.typography";

export const ColumnsContainer = styled.div`
  display: flex;
  column-gap: 175px;
`;

export const Column = styled.div`
  display: flex;
  flex-direction: column;
  gap: 35px;
`;

export const IndividualField = styled.div`
  display: flex;
  flex-direction: column;
  gap: 4px;
`;

export const FieldName = styled.div`
  font-size: ${fontSize.b1};
  font-family: ${baseFontFamily};
  line-height: 28px;
`;

export const FieldValue = styled.div`
  color: ${colors.grey100};
  font-family: ${baseFontFamily};
  font-size: ${fontSize.b2};
  line-height: 20px;
`;
