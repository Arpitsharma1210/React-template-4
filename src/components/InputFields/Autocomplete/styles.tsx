import { styled, css } from "styled-components";
import { Autocomplete } from "@mui/material";
import { colors } from "../../../theme/style.palette";

export const StyledAutoComplete = styled(Autocomplete)<{
  formatfor: "table" | "form";
}>`
  ${({ formatfor }) =>
    (formatfor === "table" &&
      css`
        width: 280px;
        background: ${colors.white25};
        border-radius: 10px !important;

        & .MuiInputBase-root {
          padding: 0px 38px 0px 9px !important;
        }

        & .MuiOutlinedInput-notchedOutline {
          border: 1px solid ${colors.grey25};
          border-radius: 10px !important;
        }
      `) ||
    (formatfor === "form" &&
      css`
        width: 100%;
        min-width: 368px !important;
        & .MuiInputBase-input {
          padding: 14px 19px;
          height: 30px;
        }
        & .MuiOutlinedInput-notchedOutline {
          border: 1px solid ${colors.grey100};
          border-radius: 6px !important;
        }
      `)}
`;
