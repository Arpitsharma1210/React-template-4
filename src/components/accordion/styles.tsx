import { styled, css } from "styled-components";
import {
  Typography,
} from "@mui/material";
import { AddCircle, RemoveCircle } from "@mui/icons-material";
import { colors } from "../../theme/style.palette";

const IconCss = () => css`
  color: ${colors.black};
  font-size: 26px;
`;

export const AddIcon = styled(AddCircle)`
  ${IconCss()}
`;

export const CloseIcon = styled(RemoveCircle)`
  ${IconCss()}
`;

export const StyledTypography = styled(Typography)<{ expanded: boolean }>`
  &.MuiTypography-root {
    order: ${({ expanded }) => (expanded ? -69 : 69)};
  }
`;
