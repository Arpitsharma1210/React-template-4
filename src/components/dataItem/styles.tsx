import { styled } from "styled-components";
import { colors } from "../../theme/style.palette";
import {
  baseFontFamily,
  fontSize,
  fontWeight,
} from "../../theme/style.typography";

export const StyledDataItem = styled.div`
  display: flex;

  & .imgContainer {
    height: 72px;
    width: 72px;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    background: ${colors.yellowGradient};
    margin-right: 16.8px;

    & img {
      width: 30px;
      height: 30px;
      display: block;
    }
  }

  & .databox {
    & .title {
      color: ${colors.grey};
      font-family: ${baseFontFamily};
      font-size: ${fontSize.b3};
      font-weight: ${fontWeight.regular};
      line-height: 16.8px;
      margin-bottom: 4px;
    }

    & .data {
      color: ${colors.grey100};
      font-family: ${baseFontFamily};
      font-size: 28px;
      font-weight: ${fontWeight.semiBold};
      line-height: 28px;
      letter-spacing: -0.28px;
      margin-bottom: 4px;
    }

    & .stat {
      & span:first-child {
        font-size: 11px;
        font-weight: ${fontWeight.bold};
      }
      & span:nth-child(2) {
        font-size: 11px;
        font-weight: ${fontWeight.regular};
        line-height: ${fontSize.b2};
        margin-right: 1px;
      }
      & span:nth-child(3) {
        font-size: 12px;
        font-weight: ${fontWeight.regular};
        line-height: ${fontSize.b2}
      }
      &.inc {
        & span:first-child {
          color: ${colors.green5};
        }
      }
      &.dec {
        & span:first-child {
          color: ${colors.red10};
        }
      }
    }
  }
`;
