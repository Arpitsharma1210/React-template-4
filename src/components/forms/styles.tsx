import styled from "styled-components";
import { colors, brand } from "../../theme/style.palette";
import {
  fontSize,
  fontWeight,
  secondaryFontFamily,
} from "../../theme/style.typography";

export const StyledCandidateHeader = styled.h3`
  color: ${brand.secondaryMain};
  font-family: ${secondaryFontFamily};
  font-size: ${fontSize.h3};
  font-weight: ${fontWeight.medium};
  line-height: 50px;
  margin: 0 0 36px 0;
`;

export const UploadButton = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;

  position: relative;
  font-size: 16px;
  width: 100%;
  border-radius: 6px;
  border: 1px solid ${colors.yellow30};
  background: ${brand.primaryMain};

  & i {
    margin-right: 18px;
  }

  & input {
    opacity: 0;
    cursor: pointer;
    position: absolute;
    left: 0;
    top: 0;
    height: 100%;
    width: 100%;
  }
`;
