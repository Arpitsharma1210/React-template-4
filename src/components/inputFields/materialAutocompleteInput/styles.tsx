import { styled, css } from "styled-components";
import { Autocomplete } from "@mui/material";
import { brand, colors } from "../../../theme/style.palette";
import { fontSize, fontWeight } from "../../../theme/style.typography";

export const StyledAutoComplete = styled(Autocomplete)<{
  formatfor: "table" | "form";
}>`
  ${({ formatfor }) =>
    (formatfor === "table" &&
      css`
        width: 280px;
        background: ${colors.white25};
        border-radius: 10px !important;

        & .MuiInputBase-root {
          padding: 0px 38px 0px 9px !important;
        }

        & .MuiOutlinedInput-notchedOutline {
          border: 1px solid ${colors.grey25};
          border-radius: 10px !important;
        }
      `) ||
    (formatfor === "form" &&
      css`
        width: 100%;
        min-width: 368px !important;
        & .MuiInputBase-input {
          padding: 14px 19px;
          height: 30px;
        }
        & .MuiOutlinedInput-notchedOutline {
          border: 1px solid ${colors.grey100};
          border-radius: 6px !important;
        }
      `)}
`;
export const StyledChipContainer = styled.div`
    display: flex;
    padding: 2px 5px 2px 10px;
    justify-content: center;
    align-items: center;
    gap: 5px;
    border-radius: 60px;
    background: ${brand.primaryMain};
    margin-right : 5px;
`

export const StyledChipLabel = styled.span`
    color : ${brand.primaryMain};
    font-size : ${fontSize.b2};
    font-weight : ${fontWeight.medium};
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    max-width: 72px;

`

export const StyledChipCloseContainer = styled.div`
    cursor : pointer;
    display : flex;
`