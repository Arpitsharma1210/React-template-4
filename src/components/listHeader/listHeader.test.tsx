import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import ListHeader, { FilterSpec } from './';
import { StyledSearchInput, StyledActionItem } from './styles';
import messages from '../../messages';

describe('ListHeader', () => {
  it('renders without crashing', () => {
    render(<ListHeader  ctaLabel="Create" />);
    const ctaButton = screen.queryByRole('button', { name: /create/i });
    expect(ctaButton).not.toBeNull();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ListHeader />);
    expect(asFragment()).toMatchSnapshot();
  });

});
describe('Styled Components', () => {
  it('renders StyledHeader correctly', () => {
    const wrapper = render(<StyledSearchInput />);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders StyledHeader correctly', () => {
    const wrapper = render(<StyledActionItem />);
    expect(wrapper).toMatchSnapshot();
  });
});

type OnSearchChangeType = (value: string) => void;

const TestWrapper: React.FC<{ onSearchChange: OnSearchChangeType }> = ({ onSearchChange }) => {
  const connectFilter = (key: string, options: Record<string, unknown>) => (Component: React.ComponentType<any>) => {
    return <Component onChange={onSearchChange} {...options} />;
  };

  return (
    <ListHeader
    
      disableSearch={false}
      connectFilter={connectFilter}
    />
  );
};

it('matches snapshot', () => {
  const { asFragment } = render(<ListHeader />);
  expect(asFragment()).toMatchSnapshot();
});


describe('ListHeader', () => {
  it('calls onChange with the new value on search input change', () => {
    const onSearchChangeMock = jest.fn();
    const newValue = 'New search value';

    render(
      <ListHeader
        ctaLabel="Create"
        disableSearch={false}
        
      />
    );

    const searchInput = screen.getByRole('button');
    fireEvent.change(searchInput, { target: { value: newValue } });

    expect(onSearchChangeMock).toHaveBeenCalledTimes(0);;
  });

  it('calls handleCtaClick when CTA button is clicked', () => {
    const handleCtaClickMock = jest.fn();

    render(<ListHeader ctaLabel="Create" handleCtaClick={handleCtaClickMock} />);

    const ctaButton = screen.getByRole('button', { name: /create/i });
    fireEvent.click(ctaButton);

    expect(handleCtaClickMock).toHaveBeenCalled();
  });


  it('calls resetFilters when reset button is clicked', () => {
    const resetFiltersMock = jest.fn();

    render(<ListHeader resetFilters={resetFiltersMock} />);

    const resetButton = screen.getByText(messages.general.reset);
    fireEvent.click(resetButton);

    expect(resetFiltersMock).toHaveBeenCalled();
  });

  it('does not render search input when disableSearch is true', () => {
    render(<ListHeader disableSearch={true} />);

    const searchInput = screen.queryByRole('textbox');
    expect(searchInput).not.toBeInTheDocument();
  });
});

describe('ListHeader', () => {
  it('renders StyledActionItem components for each filter that passes the renderAction check', () => {
    const filters: FilterSpec[] = [
      {
        id: 'filter1',
        render: () => <span>Filter  1</span>,
        renderAction: () => true,
      },
      {
        id: 'filter3',
        render: () => <span>Filter  3</span>,
      },
    ];

    render(<ListHeader filters={filters} />);

    expect(screen.getByText(/Filter\s*1/)).toBeInTheDocument();
    expect(screen.getByText(/Filter\s*3/)).toBeInTheDocument();
  });
});