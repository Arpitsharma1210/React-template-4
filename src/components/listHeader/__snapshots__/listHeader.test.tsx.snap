// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnyFh esSmtU"
  >
    <div
      class="sc-gtsqUy cAkrRM"
    />
  </div>
</DocumentFragment>
`;

exports[`Styled Components renders StyledHeader correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="sc-eCAqax lhpMWf"
      />
    </div>
  </body>,
  "container": <div>
    <input
      class="sc-eCAqax lhpMWf"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Styled Components renders StyledHeader correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlniIP ewLYee"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlniIP ewLYee"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnyFh esSmtU"
  >
    <div
      class="sc-gtsqUy cAkrRM"
    />
  </div>
</DocumentFragment>
`;
